packages/builders/src/components/Component.ts

```diff
@@ -16,30 +11,13 @@ export type AnyAPIActionRowComponent = APIActionRowComponent<APIActionRowCompone
  *
  * @typeParam DataType - The type of internal API data that is stored within the component
  */
-export abstract class ComponentBuilder<
-	DataType extends Partial<APIBaseComponent<ComponentType>> = APIBaseComponent<ComponentType>,
-> implements JSONEncodable<AnyAPIActionRowComponent>
-{
-	/**
-	 * The API data associated with this component.
-	 */
-	public readonly data: Partial<DataType>;
-
+export abstract class ComponentBuilder<Component extends AnyAPIActionRowComponent> implements JSONEncodable<Component> {
```

Comment:
I think we should rename AnyAPIActionRowComponent to just AnyComponent

Reply:
Isn't this confusing because of 
<img width="769" alt="image" src="https://github.com/user-attachments/assets/8f5c6b88-d142-4ac9-be58-7351d989805c">

Reply:
Okay, I made a couple of changes and moved those to a different file.